neuro_evolution_ctrnn/optimizer/optimizer_mu_lambda.py

```diff
@@ -65,35 +68,82 @@ def fct_mutation_learned(ind1):
-        return algorithms.eaMuPlusLambda(toolbox=self.toolbox, ngen=number_generations)
+        verbose = __debug__
+        gen = None
+        current_seed = None
+
+        for gen in range(self.initial_generation, number_generations):
+            self.record_individuals(self.population)
+            extra = []
+            if self.hof.items:
+                extra = list(map(self.clone, random.sample(self.population, self.conf.extra_from_hof)))
+            offspring = varOr(self.population + extra, self, self.conf.lambda_, 1 - self.conf.mutpb,
+                              self.conf.mutpb)
+
+            for ind in offspring:
+                # just a little extra for when we later want to analyze the hof manually
+                ind.generation = gen
+
+            for ind in self.population:
+                del ind.fitness.values
+            candidates = self.population + offspring
+
+            nevals, total_steps, current_seed, candidates = self.evaluate_candidates(candidates)
+            if self.hof is not None:
+                self.hof.update(offspring)
+
+            record = self.stats.compile(candidates) if self.stats is not None else {}
+            self.population[:] = self.select(candidates, self.conf.mu)
+
+            validation_fitness, validation_steps = self.validation_runs(self.population)
+            self.logbook.record(gen=gen, nevals=nevals, steps=total_steps + validation_steps,
+                                val_fit=validation_fitness, **record)
+            if verbose:
+                print(self.logbook.stream)
+            self.write_checkpoint(data=dict(generation=gen, halloffame=self.hof, population=self.population,
+                                            logbook=self.logbook, last_seed=current_seed, strategy=None,
+                                            recorded_individuals=self.recorded_individuals))
+        self.final_checkpoint_data = dict(generation=gen, halloffame=self.hof, population=self.population,
+                                          logbook=self.logbook, last_seed=current_seed, strategy=None,
+                                          recorded_individuals=self.recorded_individuals)
+        return self.logbook
+
+    @classmethod
+    def strip_strategy_from_population(cls, population, strategy_parameter_per_gene=False):
+        if len(population) == 0:
+            return population
+        return list(np.array(population)[:,
+                    :-(cls.get_number_genes_for_strategy(len(population[0]), strategy_parameter_per_gene))])
+
+    def get_number_genes_for_strategy(self, individual_size):
+        return individual_size // 2 if self.conf.strategy_parameter_per_gene else 2
+
+    @classmethod
+    def strip_strategy_from_individual(cls, individual, strategy_parameter_per_gene=False):
+        return list(
+            np.array(individual)[:-(cls.get_number_genes_for_strategy(len(individual), strategy_parameter_per_gene))])
```

Comment:
DRY. Kann die Methode vielleicht sogar ganz entfernt werden?

```suggestion
    def strip_strategy_from_individual(self, individual):
        return self.strip_strategy_from_population([individual])[0]
```